File: src/main/java/io/mths/golf/ISBN.java
package io.mths.golf;

interface ISBN {
    static void main(String[] isbns) {
        for (var isbn : isbns) {
            int checkDigit = 0, times = 10, digit;

            for (var character : isbn.getBytes())
                if (character != 45)
                    checkDigit += (character - 48) * times--;

            digit = (11 - checkDigit % 11) % 11;

            System.out.println(
                    isbn + (char) (digit > 9 ? 88 : 48 + digit)
            );
        }
    }
}

File: src/main/kotlin/io/mths/golf/isbn/ISBN.kt
package io.mths.golf.isbn

fun main(a: Array<String>) {
    for(b in a)
        println(
            b + ((11-b.replace("-","")
                .foldIndexed(0){i,r,c-> r+c.digitToInt()*(10-i) } % 11) % 11)
                .let{ if(it>9)'X' else it.digitToChar() }
        )
}

// fun main(a:Array<String>){for(b in a)println(b+((11-b.replace("-","").foldIndexed(0){i,r,c->r+c.digitToInt()*(10-i)}%11)%11).let{if(it>9)'X' else it.digitToChar()})}